server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1422,16 +1423,12 @@ public static IndexMetadata fromXContent(XContentParser parser) throws IOExcepti
             if (parser.currentToken() == XContentParser.Token.START_OBJECT) {  // on a start object move to next token
                 parser.nextToken();
             }
-            if (parser.currentToken() != XContentParser.Token.FIELD_NAME) {
-                throw new IllegalArgumentException("expected field name but got a " + parser.currentToken());
-            }
+            XContentParserUtils.ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.currentToken(), parser);
```

Comment:
I guess it's fine to throw here a `ParsingException` instead of `IllegalArgumentException`?

Reply:
I think that's fine and maybe even preferable as it gives a much easier to interpret error message by including some bytes around the broken section in the message.